src/dali/plugin/input/rest/kraken.py

```diff
@@ -0,0 +1,414 @@
+        api_secret: str,
+        native_fiat: str,
+        username: Optional[str] = None,
+        thread_count: Optional[int] = __DEFAULT_THREAD_COUNT,
+        use_cache: Optional[bool] = True,
+    ) -> None:
+        self.__api_key = api_key
+        self.__api_secret = api_secret
+
+        # We will have a default start time of July 27th, 2011 since Kraken Exchange officially launched on July 28th.
+        super().__init__(account_holder, datetime(2011, 7, 27, 0, 0, 0, 0), native_fiat, thread_count)
+        self.__username = username
+        self.__logger: logging.Logger = create_logger(f"{self.__EXCHANGE_NAME}")
+        self.__timezone = pytz.timezone('UTC')
+        self._initialize_client()
+        self._client.load_markets()
+        self.baseId_to_base = {value[_BASE_ID]: value[_BASE] for key, value in self._client.markets_by_id.items()}
+        self.baseId_to_base.update({'BSV': 'BSV'})
+        self.use_cache = use_cache
+
+    def exchange_name(self) -> str:
+        return self.__EXCHANGE_NAME
+
+    def plugin_name(self) -> str:
+        return self.__PLUGIN_NAME
+
+    def _initialize_client(self) -> kraken:
+        return kraken(
+            {
+                "apiKey": self.__api_key,
+                "enableRateLimit": True,
+                "secret": self.__api_secret,
+            }
+        )
+
+    @property
+    def _client(self) -> kraken:
+        super_client: Exchange = super()._client
+        if not isinstance(super_client, kraken):
+            raise Exception("Exchange is not instance of class kraken.")
+        return super_client
+
+    def _get_process_deposits_pagination_detail_set(self) -> Optional[AbstractPaginationDetailSet]:
+        pass
+
+    def _get_process_withdrawals_pagination_detail_set(self) -> Optional[AbstractPaginationDetailSet]:
+        pass
+
+    def _get_process_trades_pagination_detail_set(self) -> Optional[AbstractPaginationDetailSet]:
+        pass
+
+    def _process_gains(
+        self,
+        in_transactions: List[InTransaction],
+        out_transactions: List[OutTransaction],
+    ) -> None:
+        pass
+
+    def _gather_api_data(self):
+        import pickle
+        import os
+
+        cache_exists = os.path.exists(self.__CACHE_FILE)
+        if self.use_cache and cache_exists:
+            with open(self.__CACHE_FILE, 'rb') as file:
+                something = pickle.load(file)
+                return something
+
+        # get initial trade history to get count
+        index: int = 0
+        trade_history = self._client.private_post_tradeshistory(params={_OFFSET: index})
+        count: int = int(trade_history[_RESULT][_COUNT])
+        trade_history = dict()
+        while index < count:
+            trade_history.update(self._process_trade_history(index))
+            index += _TRADE_RECORD_LIMIT
+
+        ledger = self._client.private_post_ledgers(params={_OFFSET: index})
+        index = 0
+        count = int(ledger[_RESULT][_COUNT])
```

Comment:
Same as above. You can rewrite this as a single line, and avoid reusing `ledger`.